ext/filter/logical_filters.c

```diff
@@ -360,7 +360,7 @@ void php_filter_float(PHP_INPUT_FILTER_PARAM_DECL) /* {{{ */
 
 	if (decimal_set) {
 		if (decimal_len != 1) {
-			php_error_docref(NULL, E_WARNING, "Decimal separator must be one char");
+			zend_value_error("%s(): \"decimal\" option must be one character long", get_active_function_name());
```

Comment:
I don't think we should adopt function name prefix as a general pattern. The function name is part of the trace and people throwing exceptions in userland do not generally include such a prefix -- at least not that I'm aware of.

Reply:
I consider this as a special case of an argument error where the only reason we don't use `zend_argument_value_error()` is that its message format is inconvenient. I believe it would still be nice to follow its format as much as possible, that's why I added the prefix. So would you prefer a different message format? If you still think we should revise these messages, I can fix this, and prepare a follow-up PR, because we already have many similar error messages.

Reply:
Under what circumstances would you *not* add the prefix to a message?

Reply:
Personally, only when an error condition is not related to an argument. But I admit, the current format is a bit off from the usual style with the prefix. In the same time though, I like it this way, it makes it easier to find the source of the problem. If you have concerns, we could also discuss in chat what other people think about this format. :)

Reply:
Okay, let's go with this for now.